### PR TITLE
implement parser for tx data

### LINKED ISSUE #272: Simplify access to transaction properties
* **I'm submitting a ...**
  - [ ] bug report
  - [x] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

* **What is the current behavior?**

from the release 5 all the transactions are provided as `TxObject` (#238 ) that provides a common structure for transaction representation. This makes access to properties of a transaction consistent in any scenarios but it requires a knowledge of the structure and nesting of a tx.

A normal signed transaction has the following structure:

```
signed_tx (
    TAG
    Version
    spend_tx (
        recipient_id
        sender_id
        amount
        payload
        ...
    )
    Signatures    
)
``` 

using the api the above structure will be represented by a `TxObject` where to access for example the `payload` field you will have to do it like:
```
tx.data.tx.data.payload
```
that is the "path" to the `payload` field

With a meta transaction (from ga) there are 4 levels of nesting:
```
signed_tx (
    meta_tx(
        signed_tx(
            spend_tx (
                ...
            )
        )
    )    
)
``` 
that means that to access the payload field  you would have to write:
```
tx.data.tx.data.tx.data.tx.data.payload
```
* **What is the expected behavior?**

Provide a functionality that allows to access the data using a convenient way, for example 
```
tx.get_field_value("payload") 
```
that will retrieve the value of the field independently from its positions in the tx strucutre. 
  
* **Please tell us about your environment:**
  
  - SDK Version: v6.0.0
  - Python version: v3.7.0

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. forum, telegram, etc)

### PR DESCRIPTION
**Pull request by noandrea**
closes #272

### CODE REVIEW

**Comment by shekhar-shubhendu** [comment]
@noandrea have the lgtm issues been resolved

File: aeternity/transactions.py [hunk 1]
```diff
@@ -393,21 +393,37 @@ def __repr__(self):
 class TxObject:
     """
     This is a TxObject that is used throughout the SDK for transactions
-    It contains all the info associated to a transaction like transaction data,transaction hash, etx
-
+    It contains all the info associated to a transaction
```
Comment (shekhar-shubhendu) [inline aeternity/transactions.py]: to a/ with a

File: docs/ref/txobject.rst [hunk 2]
```diff
@@ -1,7 +1,87 @@
...
+
+::
+
+  tx_object = node_cli.spend(sender, recipient, "100AE")
+  # access the recipient_id for any spend transaction
+  tx_object.get("recipient_id")
+
+Metadata
+--------
+Metadata are special informations that are not part of the transaction itself but my be generated
```
Comment (shekhar-shubhendu) [inline docs/ref/txobject.rst]: my/may

File: docs/ref/txobject.rst [hunk 3]
```diff
@@ -1,7 +1,87 @@
...
+
+- ``min_fee`` minimum fee for a transaction, this value is always calculaed and can be used to 
+  evaluate the actual fee used for the transaction.
+- ``contract_id`` the id of a contract, only present when deploying a new contract, it has the form ``ct_....``.
+- ``salt`` the random generated salt to prepare the ``commitment_id`` of a name pre-claim transaction. The 
+  salt must be used then to prepare a claim transaction.
+
+TxObject data fields
+--------------------
+here is the complete list of transactions and available fields:
```
Comment (shekhar-shubhendu) [inline docs/ref/txobject.rst]: here/Here

File: docs/ref/txobject.rst [hunk 4]
```diff
@@ -1,7 +1,87 @@
...
+
+Metadata
+--------
+Metadata are special informations that are not part of the transaction itself but my be generated
+ad a additional output while creating or parsing a transaction, in particular metadata fields are:
+
+- ``min_fee`` minimum fee for a transaction, this value is always calculaed and can be used to 
+  evaluate the actual fee used for the transaction.
+- ``contract_id`` the id of a contract, only present when deploying a new contract, it has the form ``ct_....``.
+- ``salt`` the random generated salt to prepare the ``commitment_id`` of a name pre-claim transaction. The
```
Comment (shekhar-shubhendu) [inline docs/ref/txobject.rst]: it has the form/it starts with

File: docs/ref/txobject.rst (hunk 2 above)
Comment (shekhar-shubhendu) [inline docs/ref/txobject.rst]: is/are
are/is

File: docs/howto/amounts.rst [hunk 5]
```diff
@@ -0,0 +1,58 @@
+=======
+Amounts
+=======
+
+The Aeternity blockchain tokens are an ERC-20 compatible token,
```
Comment (shekhar-shubhendu) [inline docs/howto/amounts.rst]: are ERC-20 compatible

File: docs/howto/amounts.rst [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+where the unit is the Aetto (1e-18).
```
Comment (shekhar-shubhendu) [inline docs/howto/amounts.rst]: is Aetto

File: docs/ref/txobject.rst (hunk 2 above)
Comment (noandrea) [reply docs/ref/txobject.rst]: metadata is a mass noun

**Pull request merged**